### PR TITLE
chore: adds changeset snapshot action

### PR DESCRIPTION
**Pull request by alessbell**
## Snapshot Releases

This PR adds a GitHub action to automate creating "snapshot" one-off releases so that Apollo community members can test out the changes on a PR before it's merged.

If an Apollo maintainer adds a comment beginning with `/release:pr` on a PR _that already contains a changeset markdown file_, the following happens:

- the PR is checked out at its latest commit
- the code is built and changeset versions the library with the following naming convention: `0.0.0-${some tag we set}-${current timestamp on CI}`, for example `0.0.0-pr-10479-20230126204300`. This is the naming convention chosen by changesets and only the middle tag value is configurable which I set to the current PR number to make it easier to associate snapshot releases to their originating PRs
- this version number is only used for publishing at the time the action is run; it is not committed to the branch or otherwise persisted
- a "snapshot" of the current PR is published at the above version number under the npm tag `pr` (no GitHub tag is created)
- a GitHub bot comments on the PR with instructions to install via e.g. `npm i @apollo/client@0.0.0-pr-10479-20230126204300`

### One minor note
At first I thought we might, but we don't actually want a per-PR npm tag as unique tags are listed above the fold on our npm page, and we never want to install via a semantic tag with snapshots, just via inscrutable version number since 1. these are throwaway releases and 2. it should be somewhat effortful to install them.

There are no other side effects of this PR, meaning our existing changesets workflows remain as they are 🎉 

I tested this action out in a demo repo and it worked really nicely 🚀 

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](../CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: .github/workflows/comment.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+name: Comment
```
Comment (jerelmiller) [inline .github/workflows/comment.yml:1]: [nit] Could we call this `PR Release` or `Snapshot Release` or something to that effect instead? "Comment" feels a bit too broad and if we want/need other comment-related workflows, my guess is we'd probably want a separate workflow for that, vs trying to mash it in here.

Thoughts?

File: .github/workflows/comment.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+jobs:
+  release_next:
+    name: release:next
+    runs-on: ubuntu-latest
+
+    if: |
+      github.repository == 'apollographql/apollo-client' &&
+      github.event.issue.pull_request &&
+      (
+        github.event.sender.login == 'benjamn' ||
```
Comment (alessbell) [inline .github/workflows/comment.yml:17]: It'd be nice to use a GitHub team but it seemed like more trouble than it's worth (and I don't believe one exists that includes all/only the maintainers in our README).

**Review by jerelmiller — APPROVED** [review]
Ok this is actually super dope and I can't wait to try this. Thanks for getting this together so fast!

File: .github/workflows/comment.yml (hunk 1 above)
Comment (alessbell) [reply .github/workflows/comment.yml]: Totally agree 😁 updated!

**Review by jerelmiller — APPROVED** [review]

**Pull request merged**